src/main/java/uk/ac/ebi/eva/pipeline/io/mappers/VcfLineMapper.java

```diff
@@ -31,25 +30,21 @@
  */
 public class VcfLineMapper implements LineMapper<List<Variant>> {
 
-    private final VariantSource source;
+    private final String fileId;
+    private final String studyId;
 
     private final VariantVcfFactory factory;
 
-    public VcfLineMapper(VariantSource source) {
-        if (!VariantSource.Aggregation.NONE.equals(source.getAggregation())) {
-            throw new IllegalArgumentException(
-                    this.getClass().getSimpleName() + " should be used to read genotyped VCFs only, " +
-                            "but the VariantSource.Aggregation set to " + source.getAggregation().toString());
-        }
-        this.source = source;
+    public VcfLineMapper(String fileId, String studyId) {
+        this.fileId = fileId;
+        this.studyId = studyId;
         this.factory = new VariantVcfFactory();
     }
 
     @Override
     public List<Variant> mapLine(String line, int lineNumber) {
         assertNotNull(this.getClass().getSimpleName() + " should be used to read genotyped VCFs only " +
-                              "(hint: set VariantSource.Aggregation to NONE)",
-                      factory);
-        return factory.create(source, line);
+                        "(hint: set VariantSource.Aggregation to NONE)", factory);
```

Comment:
This assert can not fail. It was here in the previous version because the constructor could throw an exception, leaving the `factory` uninitialized.